### DISCUSSION TITLE
General

### DISCUSSION THREAD

**Discussion by keertirajmalik** [ExitTrance/KanjiMaru#40] [General]
So how's the app development going on?

**Comment by keertirajmalik** [comment]
have tried the ADB wifi?

**Reply by ExitTrance** [reply]
Hey Malik, it's going alright! Been giving all of my effort into designing a robust and scalable Firestore schema and algorithm to handle the Items that would be reviewed by a User. A big problem is that a list of items from for example the book 'Genki 1', has over 1.3k vocabulary items. A naive way of adding these items to Firestore would easily make it so that it would incur ridiculous costs. 

I have not tried adb over wifi. I have a Chromebook as well as my main Windows machine, which I'm able to run/debug my flutter apps on directly. 😄 

How is your Project 'Pocket Friend' going? Wondering how it currently looks! Any roadblocks you've come across along the way?

**Reply by keertirajmalik** [reply]
I have several things planned in my project as well but my project is not back end heavy as yours

Mine more on UI orientated as of now 😎

**Reply by keertirajmalik** [reply]
What is the total size of those genki files ?

**Reply by ExitTrance** [reply]
It's just 500 KB (in JSON Format). Saving the whole file as one huge map in Firestore might work, but I'm not sure if I'll hit the Firestore Document limit or not.

**Reply by keertirajmalik** [reply]
Why not store it in local then?

**Reply by ExitTrance** [reply]
Because I will have more than just this one list. There are so many possible vocab lists not just from books, but also exams like the JLPT, which I don't want the end product to have stored locally. 500KB might not seem much, but times that by 20-30, and you'd end up at a hefty amount of unnecessary bloat imo. This 'optimising' (if you can call it that) is premature, but that's my personality I guess. I'll probably just separate the book into sections (which are defined in the JSON), and see how it goes. 

If I get this part right, the custom user lists will also work flawlessly, so there is that as well. 😄 At the end of the day, I'm trying to fine tune it to my needs, which might make this project look a bit weird haha.

**Reply by Oran** [reply]
3q

**Reply by ExitTrance** [reply]
> 3q

不用谢同学. 一四ok

**Reply by keertirajmalik** [reply]
What is that?
Also do you have GitHub mobile it's quite handy?

**Reply by ExitTrance** [reply]
'3q' (三q) is just a Chinese meme (can also be a Japanese meme haha).

Never tried GitHub mobile before. Not even sure what one would use it for tbh.

**Reply by keertirajmalik** [reply]
Github comes quite handy for creating the issues commenting on those issues you can create PR on that as well

I sure you will like it give it a go for once 😎

**Reply by keertirajmalik** [reply]
https://github.com/mobile
